Attach informations from debian/changelog to compile.h.

---
commit 932fa20ce2db25e496f6aedf6f624e939f3dafa8
tree 87ebceaf3f1a47f9fafef53ad36510fab9cf3d5f
parent 05dbb1e557b392c8cef3675b4ba6497794ad4124
author Bastian Blank <waldi@debian.org> Tue, 13 Sep 2005 22:03:33 +0000
committer Bastian Blank <waldi@debian.org> Tue, 13 Sep 2005 22:03:33 +0000

 scripts/mkcompile_h |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/scripts/mkcompile_h b/scripts/mkcompile_h
--- a/scripts/mkcompile_h
+++ b/scripts/mkcompile_h
@@ -55,6 +55,19 @@ UTS_TRUNCATE="sed -e s/\(.\{1,$UTS_LEN\}
   fi
 
   echo \#define LINUX_COMPILER \"`$CC -v 2>&1 | tail -n 1`\"
+
+  if [ -e debian/changelog ]; then
+    DISTRIBUTION=$(lsb_release -is 2>/dev/null)
+    [ "$DISTRIBUTION" ] || DISTRIBUTION=Debian
+    MAINTAINER=$(dpkg-parsechangelog | sed -ne 's,^Maintainer: .[^<]*<\([^>]*\)>,\1,p')
+    VERSION=$(dpkg-parsechangelog | awk '/^Version:/ {print $$2}')
+    echo \#define LINUX_COMPILE_SYSTEM_DISTRIBUTION \"$DISTRIBUTION\"
+    echo \#define LINUX_COMPILE_SYSTEM_MAINTAINER \"$MAINTAINER\"
+    echo \#define LINUX_COMPILE_SYSTEM_VERSION \"$VERSION\"
+    if [ -e debian/official ]; then
+      echo \#define LINUX_COMPILE_SYSTEM_OFFICIAL
+    fi
+  fi
 ) > .tmpcompile
 
 # Only replace the real compile.h if the new one is different,



!-------------------------------------------------------------flip-


Use debian packaging informations in version string.

---
commit f9117a71c91d355243befb1a24bc557ea5d98e5b
tree 30e508720f0783c2a93b9103edd5e710f81f7f34
parent 932fa20ce2db25e496f6aedf6f624e939f3dafa8
author Bastian Blank <waldi@debian.org> Tue, 13 Sep 2005 22:04:14 +0000
committer Bastian Blank <waldi@debian.org> Tue, 13 Sep 2005 22:04:14 +0000

 init/version.c |   14 ++++++++++++--
 1 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/init/version.c b/init/version.c
--- a/init/version.c
+++ b/init/version.c
@@ -29,5 +29,15 @@ struct new_utsname system_utsname = {
 EXPORT_SYMBOL(system_utsname);
 
 const char linux_banner[] =
-	"Linux version " UTS_RELEASE " (" LINUX_COMPILE_BY "@"
-	LINUX_COMPILE_HOST ") (" LINUX_COMPILER ") " UTS_VERSION "\n";
+	"Linux version " UTS_RELEASE " "
+#ifdef LINUX_COMPILE_SYSTEM_OFFICIAL
+	"(" LINUX_COMPILE_SYSTEM_DISTRIBUTION " "
+	    LINUX_COMPILE_SYSTEM_VERSION ") "
+	"(" LINUX_COMPILE_SYSTEM_MAINTAINER ") "
+#else
+# ifdef LINUX_COMPILE_SYSTEM_VERSION
+	"(" LINUX_COMPILE_SYSTEM_VERSION ") "
+# endif
+	"(" LINUX_COMPILE_BY "@" LINUX_COMPILE_HOST ") "
+#endif
+	"(" LINUX_COMPILER ") " UTS_VERSION "\n";



!-------------------------------------------------------------flip-


